### PR TITLE
lby task1

### PR DESCRIPTION
**Pull request by lllbbbyyy**

### CODE REVIEW

File: solve1/main.cpp [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+    //输出路径
+    const cv::String img_out = "armor.jpg";
+
+    //读取图片并且进行判断,返回右值，减少拷贝
+    cv::Mat &&img = readImg(img_get);
+
+    //获取灰度图片
+    cv::Mat &&gray_img = normalToGray(img);
+    //获取灯条二值化图片
+    cv::Mat &&bin_img_light = grayToBin(gray_img, tv_light);
```
Comment (ZhengQiushi) [inline solve1/main.cpp:390]: why not use the function `hsvToBin` for color filtering?

File: solve1/main.cpp [hunk 2]
```diff
@@ -0,0 +1,413 @@
+#include <opencv2/opencv.hpp>
+#include <opencv2/imgproc/types_c.h>
+#include <iostream>
+#include <ctime>
+#include <functional>
+#include <vector>
+#include <cmath>
+
+//定义了一个矩形列表，每一维度定义了一个矩形，矩形的每一维度是顺时针方向的四个点
+typedef std::vector<std::vector<cv::Point2f>> RectAreaList;
```
Comment (ZhengQiushi) [inline solve1/main.cpp:10]: why not use the `cv::Rect` instead of `std::vector<cv::Point2f>`?

File: solve1/main.cpp [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+    RectAreaList match_rect;
+    while (i < rect_list.size())
+    {
+        if (i == rect_list.size() - 1)
+        {
+            break;
+        }
+        auto p1 = rectCenter(rect_list[i]);
+        auto p2 = rectCenter(rect_list[i + 1]);
+        //因为是排好序的，所以如果p1.x大的话说明是换行的情况，如果是相邻的话则检测中心区域
```
Comment (ZhengQiushi) [inline solve1/main.cpp:305]: please add more explanation about this part of code.

File: solve1/main.cpp [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+        if (p1.x < p2.x && judgeWhite(bin_img, midPoint(p1, p2)))
```
Comment (ZhengQiushi) [inline solve1/main.cpp:314]: please add more explanation about this part of code.

File: solve1/main.cpp [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+    //筛选轮廓
+    RectAreaList &&selected_rect_area = selectRectArea(rect_area, selectMethod);
+    showDemo("筛选轮廓", gray_img, selected_rect_area, img_size);
+    //对轮廓进行排序
+    std::sort(selected_rect_area.begin(), selected_rect_area.end(), rectCmp);
+    //获得配对以后的矩形框
+    RectAreaList &&match_rect_list = matchRect(bin_img_num, selected_rect_area);
+    showDemo("灯条与配对结果", gray_img, match_rect_list, img_size);
+    //获得装甲板矩形框
+    RectAreaList &&plate_list = getPlateList(match_rect_list);
```
Comment (ZhengQiushi) [inline solve1/main.cpp:409]: maybe you can define  class `armor` and class  `light` to record these information.

**Review by ZhengQiushi — COMMENTED** [review]
Nice coding style! It's pretty clear for others to read!

File: solve1/main.cpp (hunk 1 above)
Comment (lllbbbyyy) [reply solve1/main.cpp:390]: I tried it, but it didn’t perform very well

File: solve1/main.cpp (hunk 2 above)
Comment (lllbbbyyy) [reply solve1/main.cpp:10]: oh,u r right,I didn’t realize this before

**Pull request closed without merging**